testsuite/features/step_definitions/command_steps.rb

```diff
@@ -179,6 +179,17 @@
   $command_output, _code = get_target('server').run(command)
 end
 
+When(/^I use spacewalk-common-channel to add all "([^"]*)" channels$/) do |os_product_version|
+  raise ScriptError, "Synchronization error, version type #{os_product_version} in #{product} product not found" unless CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, os_product_version)
```

Comment:
Just a nitpick.

You can probably save the return of 

`CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, os_product_version)`

to a variable and use it both in this guard clause and in the following `each` loop

Reply:
Good catch! It improved code readability.
Done